src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -137,7 +138,16 @@ trait Contexts { self: Analyzer =>
   }
 
   def rootContext(unit: CompilationUnit, tree: Tree = EmptyTree, throwing: Boolean = false, checking: Boolean = false): Context = {
-    val rootImportsContext = rootImports(unit).foldLeft(startContext)((c, sym) => c.make(gen.mkWildcardImport(sym), unit = unit))
+    val rootImportsContext = rootImports(unit).foldLeft(startContext) { (c, sym) =>
+      if (sym.isPackage) c.make(gen.mkWildcardImport(sym), unit = unit)
+      else {
+        val universals = sym.typeSignature.members.filter(isUniversalMember).map(_.name).toSet
+        val sels = universals.iterator
+          .map(ImportSelector.mask).toList
+          .appended(ImportSelector.wild)
+        c.make(gen.mkImportFromSelector(sym, sels), unit = unit)
```

Comment:
This hurts compiler performance markedly. I haven't investigated profiles, so I'm not sure whether the cost is creating this `Import` or the overhead of subsequent symbol lookups using it rather than the simple wildcard.

Could we instead make `ImportInfo.importedSymbol` exclude these, perhaps conditionally on `isRootImport`?

It already filters its results through `isUnimportable` which excludes constructors and `private[this]`. There is also logic in there that purports to stop all imports of members of `AnyRef`, but was disabled fully, maybe unintentionally, in 2b4cd6c11016d046affb1d4ae787f85c6be1deec.

Reply:
Definitely for minor values of major.

Reply:
I was more worried about compiler performance when merging this one: https://github.com/scala/scala/pull/8538/files#diff-62ccdfe29e58b8e9f06ed66d4cfc88a5R590-R593

Reply:
Basically `Any` is not your friend. And how could it be?

Reply:
JZ's improvement in the linked PR. I'll follow up by running benchmarks and then by putting LR's mind at ease, if I am able.